docs/ja-JPN/package.md

```diff
@@ -0,0 +1,95 @@
+<div align="center">
+  <a href="http://github.com/oh-my-fish/oh-my-fish">
+    <img width=120px  src="https://cloud.githubusercontent.com/assets/8317250/8510172/f006f0a4-230f-11e5-98b6-5c2e3c87088f.png">
+  </a>
+</div>
+
+<br>
+
+<p align="center">
+  <a href="../en-US/Packages.md">English</a> &bull;
+  <a href="../zn-CN/Packages.md">简体中文</a> &bull;
+  <a href="../ru-RU/Packages.md">Русский</a> &bull;
+  <b>日本語</b>
+</p>
+
+# Creating
+
+パッケージを作ることについて学びたいのなら、あなたのシェルに`hello_world`コマンドを与える新しいパッケージを作ってみましょう。パッケージの名前は小文字や単語同士をつなぐハイフンしか使えません。
+
+Oh My Fishはあなたのためのパッケージ構成をスキャフォールド生成します。`omf new`コマンドを使ってください:
+
+```fish
+$ omf new pkg hello_world
+```
+
+> テーマには `omf new theme my_theme_name` を使ってください。
+
+ユーティリティは現在いるディレクトリから新しく生成されたパッケージに変わります:
+
+```
+$ ls -l
+  README.md
+  hello_world.fish
+  completions/hello_world.fish
+```
+
+いつでも`README.md`にあなたのパッケージがどんな働きをするのかを加えることができます。 また[auto completion](http://fishshell.com/docs/current/commands.html#complete) をよく読んだ上で、適切なタイミングかつ注意した上であなたのユーティリティにそれを加えてください。
+
+`hello_world.fish`は１つの機能を定義します:
+
+```fish
+function hello_world -d "Prints hello world"
+  echo "Hello World!"
+end
+```
+
+あなたのパッケージの中のどの機能もそれらが記述されたファイル内で宣言されていないといけません。
+オンデマンドで機能をロードしたり、使われない機能がローディングされるのを避けたりするオートローディングのメカニスムによって必須となります。
+
+fishはプライベートスコープを持たないということを覚えておいてください。なので、もしあなたのパッケージを機能毎に分割する必要がある場合は次のようにプライベートスコープをプレフィックシングすることを考えてみてください: `__hello_world.my_extra_function`, すでに存在する値への再定義(global namespace pollution)を避けるためです.
+
```

Comment:
パッケージ内の各関数は、functionsディレクトリ下にファイルを作成する必要があります。起動時に未使用の関数をロードするのを避け、オンデマンドで関数をロードします。

fishにはプライベートスコープがないことに注意してください。パッケージを関数に分割する必要がある場合は、関数名の前にパッケージ名（hello_world_print_help等）のような固有の名前を付けないでください。名前空間を汚さないためには、プライベート関数の前に2つのアンダースコア（__function_name_print_help等）を付けることを検討してください。

 Hooks

Oh My Fish provides a "hooks" system that allows you to write scripts for your package that run when other interesting events occur. Packages can use these hooks to provide advanced installation, custom resource management, etc. Hooks are ordinary Fish scripts named after the event they are triggered by. Most hooks reside in a hooks directory inside a package's project directory.

Hooks that are called at startup time (init.fish and key_bindings.fish) can slow down shell startup. Be sure to avoid slow code at startup time! Also, if your package doesn't need a hook file, be sure to remove it.
The working directory inside a hook is always set to the root directory of the package. The hooks Oh My Fish currently supports are listed below:

フック

Oh My Fishは、イベント発生時に、スクリプトが実行できる "フック"システムを提供しています。フックは、高度なパッケージのインストール、カスタムリソース管理などを提供します。フックは、イベントがトリガーされた後、実行できるFishスクリプトです。通常フックは、パッケージのプロジェクトディレクトリ内のhooksディレクトリにあります。

起動時に呼び出されるフック（init.fishおよびkey_bindings.fish）は、シェルの起動を遅くする可能性があります。できるだけ、起動時に遅いコードは避けてください。また、パッケージ内にフックが必要ない場合は、必ず削除してください。
フック内の作業ディレクトリは常にパッケージのルートディレクトリに設定されます。 Oh My Fishが現在サポートしているフックは以下の通りです：

init

The init hook is run once when the shell first loads. Scripts to handle this hook should be located at init.fish at package's root directory.

Inside this hook, you can access three package-related variables:

$package: Package name
$path: Package installation path
$dependencies: Package dependencies
For example, with an init.fish script containing the following code:

echo "hello_world initialized"
you will see the line hello_world initialized at the top of the terminal when it is first opened.

Use this hook to modify the environment, load resources, autoload functions, etc. If your package does not export any function, you can still use this event to add functionality to your package, or dynamically create functions.

Init

initフックは、シェルが最初にロードされるときに一度だけ実行されます。このフックスクリプトは、パッケージのルートディレクトリのinit.fishに置く必要があります。

initフックは、3つのパッケージ関連の変数にアクセスできます：

$ package：パッケージ名
$ path：パッケージのインストールパス
$ dependencies：パッケージの依存関係

例：init.fish
echo "hello_world initialized"
初回起動時、hello_worldがターミナル上部に表示されます。

このフックは、環境変更、リソース、オートロード機能などが行えます。関数をエクスポートしない場合でも、nitフックでパッケージに機能を追加したり、機能を動的に作成したりできます。

key_bindings

If your package or theme need to use key bindings, be sure to set them up in the key_bindings hook. Key binding scripts must be located at key_bindings.fish at package's root directory. In this hook you can freely use the bind command to define custom key bindings.

Themes can define key bindings too! Oh My Fish will reload key bindings when you switch themes.

key_bindings

key_bindingsフックに、パッケージまたはテーマのキーバインドが設定できます。キーバインディングスクリプトは、パッケージのルートディレクトリのkey_bindings.fishに配置する必要があります。このフックでは、bindコマンドを利用して、カスタムキーバインディングを定義することができます。

テーマでも重要なキーバインディングが定義でき、テーマを変更の際、キーバインドを再ロードします。

install

The install hook is triggered when a package is first installed. Scripts for this hook must be located at hooks/install.fish.

Inside this hook, you can access two package-related variables:

$package: Package name
$path: Package installation path
This hook is useful for downloading additional resources, setting up Git submodules, or installing third-party dependencies like Bash scripts.

インストール

インストールフックは、パッケージがインストールされた際にトリガーされます。このフックのスクリプトは、hooks / install.fishに置く必要があります。

このフックは、2つのパッケージ変数にアクセスできます。

$ package：パッケージ名
$ path：パッケージのインストールパス
このフックは、追加のリソースをダウンロード、Gitサブモジュールの設定、Bash等の依存関係のあるサードパーティをインストールするのに便利です。

update

As you might have guessed, the update hook is triggered for a package after it is updated. Scripts for this hook must be located at hooks/update.fish.

Inside this hook, you can access two package-related variables:

$package: Package name
$path: Package installation path
This hook is useful for updating Git submodules or checking for new versions of third-party dependencies.

更新

更新フックはパッケージ更新後に対してトリガーされます。このフックのスクリプトは、hooks / update.fishに置く必要があります。

このフックは、2つのパッケージ変数にアクセスできます。

$ package：パッケージ名
$ path：パッケージのインストールパス
このフックは、Gitサブモジュールを更新、サードパーティの依存関係の最新バージョンのチェック等に便利です。


uninstall

The uninstall hook will be triggered before a package is removed via omf remove <pkg>. Scripts for this hook must be located at hooks/uninstall.fish.

Inside this hook, you can access two package-related variables:

$package: Package name
$path: Package installation path
Packages can use this hook to clean up custom resources, etc.

Note: for backwards-compatibility, uninstall hooks will also be run if they are located at uninstall.fish in the package root.

アンインストール

omf remove <pkg>でパッケージを削除する前に、アンインストールフックがトリガーされます。このフックスクリプトは、hooks / uninstall.fishに置く必要があります。

このフックは、2つのパッケージ変数にアクセスできます。

$ package：パッケージ名
$ path：パッケージのインストールパス
このフックはパッケージのカスタムリソースなどをクリーンアップすることができます。

注意：下位互換性のため、アンインストールフックはパッケージルートのuninstall.fishにある場合にも実行されます。


Make it public

The official registry of public packages is managed in the oh-my-fish/packages-main repository. See the README of that repository for instructions on how to add your package to the official package database.

公開

公開するパッケージの公式登録は、oh-my-fish / packages-mainリポジトリで管理されています。公式のデータベースにパッケージを追加する方法については、該当するリポジトリのREADMEを参照してください。